src/python_testing/TC_MWOCTRL_2_4.py

```diff
@@ -0,0 +1,102 @@
+from mobly import asserts
+
+# This test requires several additional command line arguments
+# run with
+# --int-arg PIXIT_ENDPOINT:<endpoint>
+
+
+class TC_MWOCTRL_2_4(MatterBaseTest):
+
+    async def read_mwoctrl_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.MicrowaveOvenControl
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    def desc_TC_MWOCTRL_2_4(self) -> str:
+        return "[TC-MWOCTRL-2.4] WATTS functionality with DUT as Server"
+
+    def steps_TC_MWOCTRL_2_4(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep(2, "Read the SupportedWatts attribute"),
+            TestStep(3, "Read the SelectedWattIndex attribute"),
+            TestStep(4, "Send the SetCookingParameters command"),
+            TestStep(5, "Read and verify the SelectedWattIndex attribute"),
+        ]
+        return steps
+
+    def pics_TC_MWOCTRL_2_4(self) -> list[str]:
+        pics = [
+            "MWOCTRL.S",
+        ]
+        return pics
+
+    @async_test_body
+    async def test_TC_MWOCTRL_2_4(self):
+
+        endpoint = self.user_params.get("endpoint", 1)
+
+        self.step(1)
+        attributes = Clusters.MicrowaveOvenControl.Attributes
+        feature_map = await self.read_mwoctrl_attribute_expect_success(endpoint=endpoint, attribute=attributes.FeatureMap)
+        features = Clusters.MicrowaveOvenControl.Bitmaps.Feature
+        commands = Clusters.Objects.MicrowaveOvenControl.Commands
+
+        feature_map = await self.read_mwoctrl_attribute_expect_success(endpoint=endpoint, attribute=attributes.FeatureMap)
+
+        only_watts_supported = feature_map == features.kPowerInWatts
+
+        if not only_watts_supported:
+            logging.info("PowerInWatts is not supported so skipping the rest of the tests.")
+            # Skipping all remainig steps
+            for step in self.get_test_steps(self.current_test_info.name)[self.current_step_index:]:
+                self.step(step.test_plan_number)
+                logging.info("Test step skipped")
+            return
+
+        self.step(2)
+        supportedWattsList = await self.read_mwoctrl_attribute_expect_success(endpoint=endpoint, attribute=attributes.SupportedWatts)
+        asserts.assert_true(len(supportedWattsList) > 0, "SupportedWatts list is empty")
+
+        self.step(3)
+        selectedWattIndex = await self.read_mwoctrl_attribute_expect_success(endpoint=endpoint, attribute=attributes.SelectedWattIndex)
+        logging.info("SelectedWattIndex is %s" % selectedWattIndex)
+        asserts.assert_true(selectedWattIndex >= 0 and selectedWattIndex < len(
+            supportedWattsList), "SelectedWattIndex is out of range")
+
+        self.step(4)
+        newWattIndex = (selectedWattIndex+1) % (len(supportedWattsList)-1)
```

Comment:
test plan question - given that this is written in python now, why not just try all of them in a for loop?

Reply:
https://github.com/CHIP-Specifications/chip-test-plans/issues/3914